src/NotificationProcessor/OfferClosedProcessor.php

```diff
@@ -0,0 +1,42 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2021 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\NotificationProcessor;
+
+use Shopware\Core\Checkout\Order\Aggregate\OrderTransaction\OrderTransactionStates;
+use Shopware\Core\Framework\Context;
+
+class OfferClosedProcessor extends BaseProcessor
+{
+    public function process()
+    {
+        $orderTransaction = $this->getOrder()->getTransactions()->first();
```

Comment:
can there be more than 1 transaction in case of multiple payment attempt?

Reply:
Yes, there can. In the case an order fails and then succeeds, the order still has multiple transactions, but I did a few data dumps and I found that the OTs are listed in reverse chronological order.

Reply:
Could you please create a ticket with low prio to discuss the possibility of processing each notification while connecting them to a transaction?

I don't know if there is any added value but maybe based on pspreference we can connect the transaction with the notification and therefore we only need to process the notifications which are connected to the last transaction.
I am not sure though if we have pspreference for each attempt - I think for redirectShoppers it's not necessary to have one.

Let's discuss it under a new ticket but probably it won't change anything :)